internal/diagnose/config_validator.go

```diff
@@ -71,6 +72,11 @@ func (c *ConfigValidator) Validate(ctx context.Context) error {
 	postEvent := func() error {
 		if err = c.client.Events.CreateEventWithContext(ctx, accountID, evt); err != nil {
 			log.Debug(err)
+
+			if e, ok := err.(*nrErrors.PaymentRequiredError); ok {
```

Comment:
I see we don't have any test for the config validator (debt), I'm a bit sorry to ask since the addition here is small, but this feels like it should be unit tested too.
In addition, it took me a bit of time to understand the value of this addition here, versus just returning the error below.